libraries/AP_HAL_SITL/sitl_rangefinder.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+  SITL handling
+
+  This simulates a rangefinder
+
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+
+#include "AP_HAL_SITL.h"
+#include "AP_HAL_SITL_Namespace.h"
+#include "HAL_SITL_Class.h"
+
+#include <SITL/SITL.h>
+
+extern const AP_HAL::HAL& hal;
+
+using namespace HALSITL;
+
+// TODO Improve that to not use analog
+/*
+  setup the rangefinder with new input
+ */
+void SITL_State::_update_rangefinder(float range_value)
+{
+    float altitude = range_value;
+    if (range_value == -1) {
+        altitude = _sitl->height_agl;
+    }
+
+    // sensor position offset in body frame
+    Vector3f relPosSensorBF = _sitl->rngfnd_pos_offset;
+
+    // adjust altitude for position of the sensor on the vehicle if position offset is non-zero
+    if (!relPosSensorBF.is_zero()) {
+        // get a rotation matrix following DCM conventions (body to earth)
+        Matrix3f rotmat;
+        _sitl->state.quaternion.rotation_matrix(rotmat);
+        // rotate the offset into earth frame
+        Vector3f relPosSensorEF = rotmat * relPosSensorBF;
+        // correct the altitude at the sensor
+        altitude -= relPosSensorEF.z;
+    }
+
+    float voltage = 5.0f;
+    if (fabsf(_sitl->state.rollDeg) < 90 &&
+            fabsf(_sitl->state.pitchDeg) < 90) {
```

Comment:
Ok I will try to remember that next time ! 

About the fabsf, it should stay fabs for now .  I don't submit the change from sitl state from double to float .